### ISSUE TITLE
double height text doesn't line-wrap properly

### ISSUE THREAD

**Issue by ClaireCJS** [microsoft/terminal#15681]
### Steps to reproduce

Create double-height text and then use it to write a line longer than the width of the current console window/pane

### Expected Behavior

that it would wrap into another double-height line

### Actual Behavior

It actually breaks the double-height line into two and ruins the effect, and then wraps the remainder twice as single height lines

![image](https://github.com/microsoft/terminal/assets/789591/93c6b66b-0857-4f94-aff8-12d975e7c0ac)

Perhaps the wrapped text of a "big" line can be buffered and if it matches the wrapped text of the next line and that line is also "big", then a new "big" line can be created with the wrapped text and inserted after?

**Comment by ClaireCJS** [comment]
Here's another more carefully constructed example, and a rendition of what in my opinion I think should have happened:

![image](https://github.com/microsoft/terminal/assets/789591/8e76688f-c854-4864-a76a-42f9ace23f8d)

**Comment by j4james** [comment]
> Perhaps the wrapped text of a "big" line can be buffered and if it matches the wrapped text of the next line and that line is also "big", then a new "big" line can be created with the wrapped text and inserted after?

Yeah. this is something I had hoped we might be able to do one day. I don't think it even matters if the lines don't match. As long as you've got a top and bottom double-height pair, I think that should be enough to trigger a combined wrap.

That said, there are a bunch of edge cases which will likely make this more complicated than it seems at first glance. So as an interim solution we might want to consider just truncating any double height lines when they're resized. That at least would be a visual improvement, although the downside is that you're losing content in the process.

**Comment by ClaireCJS** [comment]
I guess under-the-hood it's more complicated than "just store what you truncate and later spit it out" :)

You're right, it shouldn't matter if the lines match.  If someone wants mismatched lines, that's their prerogative. There's some neat/funny things that can be done with that. 

![image](https://github.com/microsoft/terminal/assets/789591/a101a154-b270-403f-82ec-1149cbca79ca)

**Comment by j4james** [comment]
Just to be clear here, when you're referring to the text wrapping, do you mean when an application outputs text that doesn't fit on one line? Or are you talking about text that was originally on one line, but was later forced to wrap by the terminal when the window was resized?

I assumed the latter, because window resizing is the terminal's responsibility. However, if you actually meant the former, then that's up to the application to handle correctly. If you've got double height text that isn't expected to fit on a single line, it's the app's responsibility to split that in two, and write out the two instances of each half with appropriate line renditions.

**Comment by ClaireCJS** [comment]
Yea i meant the first.

That seems weird that it's the app's responsibility to handle line-wrapping.

**Comment by j4james** [comment]
It's not like this is something unique to Windows Terminal though. This is a standard level 1 VT capability that has existed for decades. We're just following the standard.

And typically these attributes would be used for things like headings, which wouldn't be expected to wrap. Although it's not particularly difficult for an application to manually wrap double-height text if they really wanted to.

**Comment by ClaireCJS** [comment]
> I really liked the "truncation" hot-fix so I implemented it in #15701. I'm personally not intending to implement better alternatives (like wrapping double height rows in pairs).

Amy chance we could just truncate the 1st of the two?  that way the 2nd one can still line-wrap to normal size and no output is lost, but the double-height visual isn't perturbed either. 

Basically just, don't truncate a line if the last one was?

**Comment by ClaireCJS** [comment]
Hi!

As a follow-up, and possibly the same or a different bug, I wanted to illustrate this phenomenon with a better example that you should be able to easily reproduce yourself:


Something is happening prematurely. This isn't even really a wrap bug. Somehow, getting within ≈7 (but not always 7?) characters of the right of the screen inserts a blank line that shouldn't be there.

The more I think about this, the more I think this isn't a wrapping situation, so I think this is actually a separate bug that I should file, so I'm going to paste this into a new bug report.



```
rem Adjust terminal width to 209 columns first...
pause

set big_bot=←#4
set big_top=←#3
set message=there are 209 columns...there are 209 columns...there are 209 columns...there are 209 columns....
rem length of message is 97
echo %message%

echo -----------
echo %big_top%%message% 
echo %big_bot%%message%
echo -----------

rem This message is one character shorter:
set message=there are 209 columns...there are 209 columns...there are 209 columns...there are 209 columns...

echo -----------
echo %big_top%%message% 
echo %big_bot%%message%
echo -----------


```

![Image](https://github.com/user-attachments/assets/0bd6aac8-1fe8-4dfe-b0fa-75568aea3944)
